daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/FNFunctions.scala

```diff
@@ -706,6 +706,92 @@ case class FNLocalName1(recipe: CompiledDPath, argType: NodeInfo.Kind)
   }
 }
 
+/**
+ * Returns the namespace URI of the name of \$arg as an xs:anyURI
+ * value.
+ *
+ * If the argument is omitted, it defaults to the context item (.).
+ * The behavior of the function if the argument is omitted is
+ * exactly the same as if the context item had been passed as
+ * the argument.
+ *
+ * If the node identified by \$arg is neither an element nor an
+ * attribute node, or it is an element or attribute node whose
+ * expanded-QName is in no namespace, then the function returns
+ * the zero-length xs:anyURI value.
+ *
+ * Otherwise, the result will be the namespace URI of the
+ * expanded-QName of the node identified by \$arg returned as an
+ * xs:anyURI value.
+ *
+ * The following errors may be raised when \$arg is omitted:
+ * - If the context item is absent, dynamic error [err:XPDY002]
+ * - If the context item is not a node, type error [err:XPTY004]
+ *
+ * This function is called when 0 arguments are provided.  We
+ * treat this as if the argument passed was "." to denote self.
+ */
+case class FNNamespaceUri0(recipe: CompiledDPath, argType: NodeInfo.Kind)
+  extends RecipeOpWithSubRecipes(recipe) {
+  override def run(dstate: DState) {
+    // Hook so we can insist this is non-constant at compile time
+    dstate.fnExists()
+    // Check that we have a current node, otherwise return empty string(URI)
+    if (dstate.currentNode eq null) {
+      dstate.setCurrentValue("")
+    }
+    else {
+      // Same as using "." to denote self.
+      val value: DataValueURI = dstate.currentNode.namedQName.namespace.optURI.get
```

Comment:
I think optURI is an optional because if the namespace of an element is NoNamspace then optURI is None. According to the DFDL spec, if the namespace is NoNamespace, then this should return empty string. So we probably need a match/case here, somethign like:
```scala
val value = dstate.currentNode.namedQName.namespace.optURI match {
  case Some(uri) => uri
  case None => new URI("")
}
```
Would be good to add a test case for a no namespace schema.

Reply:
Yes, I agree if we add the above match/case, we also need test cases for both branches.

Reply:
I added a match/case and test case for a node in no namespace.